identity/identity_test.go

```diff
@@ -15,3 +20,79 @@ func TestNewIdentity(t *testing.T) {
 	assert.NotEmpty(t, i.Traits)
 	assert.NotNil(t, i.Credentials)
 }
+
+func TestMarshalExcludesCredentials(t *testing.T) {
+	i := NewIdentity(config.DefaultIdentityTraitsSchemaID)
+	credentials := map[CredentialsType]Credentials{
+		CredentialsTypePassword: Credentials{
+			ID: uuid.UUID{},
+		},
+	}
+
+	i.Credentials = credentials
+	jsonBytes, err := json.Marshal(i)
+	assert.Nil(t, err)
+	var jsonMap = map[string]json.RawMessage{}
+	err = json.Unmarshal(jsonBytes, &jsonMap)
+	assert.Nil(t, err)
+	assert.Nil(t, jsonMap["credentials"])
+	assert.Equal(t, credentials, i.Credentials, "Original credentials should not be touched by marshalling")
+}
+
+func TestMarshalExcludesCredentialsByReference(t *testing.T) {
+	i := NewIdentity(config.DefaultIdentityTraitsSchemaID)
+	credentials := map[CredentialsType]Credentials{
+		CredentialsTypePassword: Credentials{
+			Type: CredentialsTypePassword,
+		},
+	}
+	i.Credentials = credentials
+	jsonBytes, err := json.Marshal(&i)
+	assert.Nil(t, err)
+	var jsonMap = map[string]json.RawMessage{}
+	err = json.Unmarshal(jsonBytes, &jsonMap)
+	assert.Nil(t, err)
+	assert.Nil(t, jsonMap["credentials"])
+	assert.Equal(t, credentials, i.Credentials, "Original credentials should not be touched by marshalling")
```

Comment:
@aeneasr This one shows "credentials" is not set to nil.

Reply:
Ah, nice, thank you!